### DISCUSSION TITLE
Prioritized queue for loading plugins

### DISCUSSION THREAD

**Discussion by AkazaRenn** [goatcorp/Dalamud#1188] [Ideas]
Hi, I'm wondering if there's a higher priority queue regarding the order of plugins loading when the game starts. There are plugins that needs to get loaded early enough (Title Edit needs to get loaded before we get into the title screen) but there are plugins taking long before them (Market Board, Sonar). Is there a way to always make some plugins get loaded first?

**Comment by reiichi001** [comment]
Dalamud already takes this into account.

Plugins are generally loaded asynchronously, base on their `LoadPriority` number as defined in their plugin manifests. (Bigger number = higher priority)

For plugins that require a synchronous load or early load, the developers have additional options they can set, like `LoadSync`. 

In Dalamud Settings, there is also a setting called "Wait for plugins to load" which is enabled by default. When using Endpoint injection for Dalamud (check in XIVLauncher's settings - it's the default), Dalamud will pause the game process and load any plugins in early when needed. Then it loads all other plugins async. (You can also see this mentioned in your dalamud.log file!)

**Reply by AkazaRenn** [reply]
I must have missed that piece, thank you so much for the info!
